### ISSUE TITLE
[Task]: Improve the caching of lang gradle build 

### ISSUE THREAD

**Issue by azinneera** [ballerina-platform/ballerina-lang#43066]
### Description

Right now, the caching in the gradle build seems to be suboptimal. I could notice that some unchanged modules are building. For example, all the lang library builds are running even when no changes are done. This could be due to inputs and outputs of the build not being properly configured. We need to check and optimize this.

### Related area

-> Other Area

**Comment by Shadow-Devil** [comment]
Hi, I looked into it and for me the problem was that the `jvmArgumentProviders` inside the `BallerinaLangLibBuildTask` which is inherited from the `JavaExec` task is considered for up-to-date checking.
![image](https://github.com/ballerina-platform/ballerina-lang/assets/29004905/8ddbe363-586d-41bd-b4b0-e2385c99ac10)

To fix this, I used the same technique as in the Gradle docs, where they overwrite the getter method and mark it explicitly as `@Interal` to exclude it from up-to-date checking.

```java
    @Override @Internal
    List<CommandLineArgumentProvider> getJvmArgumentProviders() {
        return super.getJvmArgumentProviders()
    }
```
Reference: https://docs.gradle.org/current/userguide/build_cache.html#using_annotations

I will create a PR where I include this fix.
